### ISSUE TITLE
cmake --install installs an empty directory

### ISSUE THREAD

**Issue by tocic** [uni-algo/uni-algo#19]
When building the library according to the docs

```shell
cmake -S . -B "build" -DCMAKE_BUILD_TYPE=Release
cmake --build "build"
cmake --install "build"
```

an empty directory at `.../uni_algo/impl/doc/` is created. I think it's redundant.

Please consider [exluding it](https://discourse.cmake.org/t/prevent-creating-empty-directories-using-install-directory/4528) here.

https://github.com/uni-algo/uni-algo/blob/ef9366af5f5b807af049bff87ed4024028dbc08a/CMakeLists.txt#L103-L105

**Comment by mg152** [comment]
There is also `impl/example` dir that should not be installed. As I remember `PATTERN EXCLUDE` didn't work for me for some reason so I ignored it, will fix it later.

**Comment by mg152** [comment]
Fixed in commit 41cbce38fff025a0a2883111a2c393ea4b0d8633
It seems `PATTERN EXCLUDE` only accepts exact names and doesn't accepts paths but it's good enough.
